### PR TITLE
Move wiki pages to .md in repo

### LINKED ISSUE #16771: Move wiki pages to .md files in the repo
To make it easier for contributors to send PRs and to make it easier to get changes reviewed.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #16771

Just initial draft for move. Some of the files seems not so relevant now so we can probably drop them. Once this is merged we can improve particular page if we want to.

### CODE REVIEW

File: docs/design-sqlclient-for-ef7.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+The purpose of this design document is to define what Entity Framework requires from a SqlClient implementation to be used in EF7 with support for ASP.NET vNext on CoreCLR.
```
Comment (AndriySvyryd) [inline docs/design-sqlclient-for-ef7.md]: This file isn't relevant anymore

File: docs/getting-and-building-the-code.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+| -pack | Build and produce NuGet packages. | None |
+
+## Using Visual Studio
+
+**Important** The command line `build` (see above) must be run before using the solution with Visual Studio.
+
+The build script installs a preview .NET Core SDK. In order to make sure Visual studio (or any other IDE) is using same SDK, certain environment variables need to be set. To configure your local environment and open solution file in Visual Studio, run following command:
+
+```console
+startvs.cmd EFCore.sln
```
Comment (AndriySvyryd) [inline docs/getting-and-building-the-code.md]:
```suggestion
startvs.cmd EFCore.slnf
```

File: docs/getting-and-building-the-code.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+***
+
+# Earlier versions
+
+## Prerequisites
+
+The following prerequisites are required to work with the EF Core code:
+
+* [**Visual Studio 2017 15.8**](https://www.visualstudio.com/downloads/)
+* **.NET Core 2.2 Preview SDK** (Installed by build script automatically)
```
Comment (AndriySvyryd) [inline docs/getting-and-building-the-code.md]: 2019 16.2
3.0

File: docs/getting-and-building-the-code.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+Our tests are written using [xUnit.net](http://xunit.github.io/), and can be run with most test runners we have tried.
+Tests can be run on the command line (after build) by running `test`:
+
+```console
+test
+```
+
+***
+
+# Earlier versions
```
Comment (AndriySvyryd) [inline docs/getting-and-building-the-code.md]:
```suggestion
# Earlier versions (2.x or older)
```

File: docs/guiding-principles.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
+The purpose of this page is to list out the things that help us make decisions when designing/implementing/reviewing EF Core. This will help us make individual decisions that result in a cohesive stack.
```
Comment (AndriySvyryd) [inline docs/guiding-principles.md]: We should probably move this page to Docs and expand

File: docs/home.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Welcome to the Entity Framework Core Project Wiki
```
Comment (AndriySvyryd) [inline docs/home.md]:
```suggestion
# Welcome to the Entity Framework Core Project
```

Comment (AndriySvyryd) [inline docs/home.md]: Move relevant parts to README.md and CONTRIBUTING.md and remove this file

File: docs/nuget-package-naming.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+This document was a proposal for NuGet package naming that was eventually discarded after [much discussion](https://github.com/aspnet/Announcements/issues/42) (and [further discussion](https://github.com/aspnet/EntityFramework/issues/2508)).
```
Comment (AndriySvyryd) [inline docs/nuget-package-naming.md]: Remove this file

File: docs/security.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
+This document records the security guarantees that we make in EF7 (and later releases) and the security related guidance that developers should follow when using Entity Framework.
```
Comment (AndriySvyryd) [inline docs/security.md]: Remove EF 7 references

File: docs/visual-studio-publish-integration.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
+In Visual Studio you can **right-click -> Publish...** on an project to walk thru a deployment wizard. This wizard should help you configure/deploy all aspects of your application, including databases. There are two aspects of database deployment that apply to Entity Framework.
```
Comment (AndriySvyryd) [inline docs/visual-studio-publish-integration.md]: Should be moved to Docs and updated

File: docs/design-sqlclient-for-ef7.md (hunk 1 above)
Comment (smitpatel) [reply docs/design-sqlclient-for-ef7.md]: Deleted

File: docs/guiding-principles.md (hunk 5 above)
Comment (smitpatel) [reply docs/guiding-principles.md]: https://github.com/aspnet/EntityFramework.Docs/issues/1945

File: docs/visual-studio-publish-integration.md (hunk 9 above)
Comment (smitpatel) [reply docs/visual-studio-publish-integration.md]: https://github.com/aspnet/EntityFramework.Docs/issues/1946#apply-any-pending-migrations

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -3,4 +3,4 @@ Contributing
 
 Information on contributing to this repo is in the [Contributing Guide](https://github.com/aspnet/Home/blob/master/CONTRIBUTING.md) in the AspNetCore repo.
 
-Specific instructions on [getting and building the code](https://github.com/aspnet/EntityFrameworkCore/wiki/getting-and-building-the-code) are available on the wiki.
+Specific instructions on [getting and building the code](../docs/getting-and-building-the-code.md) are available on the wiki.
```
Comment (AndriySvyryd) [inline .github/CONTRIBUTING.md]: There is no wiki anymore

Comment (smitpatel) [reply .github/CONTRIBUTING.md]: I thought about this and then forgot to update. :(

**Pull request merged**